src/components/flowchart/index.js

```diff
@@ -66,7 +67,7 @@ export class FlowChart extends Component {
     const { chartZoom } = this.props;
     const changed = (...names) => this.changed(names, prevProps, this.props);
 
-    if (changed('visibleSidebar')) {
+    if (changed('visibleSidebar', 'visibleCode', 'visibleMetaSidebar')) {
       this.updateChartSize();
     }
```

Comment:
Yeah I've been trying to resolve this, looks like what it really needs to do is respect a minimum zoom level rather than just a width, will get back on this.

Reply:
I've now merged the view refactor work from #363 and implemented the change to account for the code panel and all is working together nicely. 

I've also added some flowchart tests to check the extents are being applied as expected with sidebars off and on.